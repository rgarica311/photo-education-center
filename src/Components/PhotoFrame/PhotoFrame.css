.eduPhotoBlur {
  filter: blur(10px);
  position: relative;
  height: 227px;
  width: 222px;
  top: 16px;
  left: 15px;
}

.eduPhoto {
  position: relative;
  height: 227px;
  width: 222px;
  top: 16px;
  left: 15px;
}

.bySpan {
  position: absolute;
  bottom: 10px;
  left: 15px;
  width: max-content;
}

.content {
  position: absolute;
  height: 227px;
  width: 222px;
  margin: 0;
  left: 15px;
  bottom: 62px;

}

.contentSpan {
  pointer-events: none;
  padding: 15px;
  height: 227px;
  width: 222px;
  left: 15px;
  color: white;
  font: 'Raleway';
  align-items: center;
  font-size: .8rem;
  font-family: 'Raleway';
  font-weight: 900;
  line-height: 1.5rem;
}

@media screen and (max-width: 600px) {
  .photoFrame {
    margin-top: 10px;
  }
}
