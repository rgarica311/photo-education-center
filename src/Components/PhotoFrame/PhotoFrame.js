import React, {Component} from 'react';
import './PhotoFrame.css'

let photoFrame = require('./photoFrame.png')

const photoFrameStyle = {
  backgroundImage: `url(${photoFrame})`,
  height: '305px',
  width: '251px',
  backgroundSize: 'cover',
  position: 'relative'

}


export default class PhotoFrame extends Component {
  constructor(props){
    super(props)
    this.state = {
      name: props.educator.name,
      photo: props.educator.photo,
      about: props.educator.about,
      displayContent: 'none',
      hover: false
    }
  }

  handleHover(e) {
    e.stopPropagation()
    console.log('hadle hover')
    this.setState({
      displayContent: 'flex'
    })
  }

  handleLeave(e) {
    e.stopPropagation()
    console.log('handle leave')
    this.setState({
      displayContent: 'none'
    })
  }

  render() {
    return (
      <>
        <div style={photoFrameStyle} className='photoFrame'>
          {
            this.state.displayContent === 'none'
              ? <img alt='edu member' onMouseLeave={e => this.handleLeave( e )} onMouseEnter={e => this.handleHover( e )} className='eduPhoto' height='227px' width='222px' src={this.state.photo}/>

            : <img alt='blurred edu member' onMouseLeave={e => this.handleLeave( e )} onMouseEnter={e => this.handleHover( e )} className='eduPhotoBlur' height='227px' width='222px' src={this.state.photo}/>

          }

            <span style={{display: this.state.displayContent}} className='contentSpan content'>
              {this.state.about}
            </span>
          <span className='bySpan'>{this.state.name}</span>
        </div>
      </>
    )
  }

}
