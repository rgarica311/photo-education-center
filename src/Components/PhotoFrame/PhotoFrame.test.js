import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PhotoFrame from './PhotoFrame';

describe(`PhotoFrame Component`, () => {
  const educator = {
    name: 'test name',
    photo: 'test photo',
    about: 'test about'
  };
  
  it('renders PhotoFrame by default', () => {
    const wrapper = shallow(<PhotoFrame educator={educator} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
