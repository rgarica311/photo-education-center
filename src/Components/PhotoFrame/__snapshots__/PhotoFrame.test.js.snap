// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoFrame Component renders PhotoFrame by default 1`] = `
<Fragment>
  <div
    className="photoFrame"
    style={
      Object {
        "backgroundImage": "url(photoFrame.png)",
        "backgroundSize": "cover",
        "height": "305px",
        "position": "relative",
        "width": "251px",
      }
    }
  >
    <img
      alt="edu member"
      className="eduPhoto"
      height="227px"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      src="test photo"
      width="222px"
    />
    <span
      className="contentSpan content"
      style={
        Object {
          "display": "none",
        }
      }
    >
      test about
    </span>
    <span
      className="bySpan"
    >
      test name
    </span>
  </div>
</Fragment>
`;
